src/server/api/routers/media-server.ts

```diff
@@ -71,22 +71,29 @@ export const mediaServerRouter = createTRPCRouter({
 const handleServer = async (app: ConfigAppType): Promise<GenericMediaServer | undefined> => {
   switch (app.integration?.type) {
     case 'jellyfin': {
+      const apiKey = findAppProperty(app, 'apiKey');
+
       const username = findAppProperty(app, 'username');
+      const password = findAppProperty(app, 'password');
 
-      if (!username) {
-        return {
-          appId: app.id,
-          serverAddress: app.url,
-          sessions: [],
-          type: 'jellyfin',
-          version: undefined,
-          success: false,
-        };
-      }
 
-      const password = findAppProperty(app, 'password');
 
-      if (!password) {
+      let api; 
+      let infoApi;
+      let sessionApi;
+
+
+      if (apiKey &&  apiKey.length > 5 ) {
+        api = jellyfin.createApi(app.url, apiKey);
+        infoApi = await getSystemApi(api).getPublicSystemInfo();
+        sessionApi = getSessionApi(api);
+      }else if (username && password){
+        api = jellyfin.createApi(app.url);
+        await api.authenticateUserByName(username, password);
+        infoApi = await getSystemApi(api).getPublicSystemInfo();
+        sessionApi = getSessionApi(api);
+      }
```

Comment:
Instead of doing this, can you extract it to a function? This way it should get easier to read